website/docs/crud.md

```diff
@@ -0,0 +1,293 @@
+Extends the class `CRUD` and implements the `get()` and `delete()` methods
+
+```python
+class CRUDDataAccess(DataAccess, CRUD):
+
+    def __init__(self, db_entity, db: SqlAlchemyDataHandlerRegistry, rule_validator: RuleValidator):
+        CRUD.__init__(self, db_entity, db, rule_validator)
+```
+`db_entity` the database entity used by this `DataAccesss`.
+
+`db` instance of the `SqlAlchemyDataHandlerRegistry` used to connect to the database.
+
+`rule_validator` is an optional parameter that uses the `RuleValidator` instance to validate data passed into `create()` and `update()` methods.
+
+#### Functions provided by `CRUDDataAccess`
+
+- `get(id: int)` overrides the function in `CRUD` class, used to get data from database for a given `id`.
+```python
+def get(self, id: int):
+```
+`id` is the id of the column we want to query.
+
+
+- `delete(id: int)` overrides the function in `CRUD` class, deletes the data for the given `id`.
+```python
+def delete(self, id: int):
+```
+`id` is the id of the column to be deleted.
+
+#### Usage
+```python
+from core_lib.data_layers.data_access.db.crud.crud import CRUD
+from core_lib.data_layers.data_access.db.crud.crud_data_access import CRUDDataAccess
+from core_lib.rule_validator.rule_validator import ValueRuleValidator, RuleValidator
+from core_lib.data_layers.data.db.sqlalchemy.base import Base
+from core_lib.data_transform.result_to_dict import result_to_dict
+
+from sqlalchemy import Column, Integer, VARCHAR, Boolean
+
+class Customer(Base):
+    __tablename__ = 'customer_data'
+
+    id = Column(Integer, primary_key=True, nullable=False)
+    name = Column(VARCHAR(length=255), nullable=False, default="")
+    email = Column(VARCHAR(length=255), nullable=False, default="")
+    active = Column(Boolean, nullable=False)
+
+
+class CustomerCRUDDataAccess(CRUDDataAccess):
+    allowed_update_types = [
+        ValueRuleValidator('name', str),
+        ValueRuleValidator('email', str),
+        ValueRuleValidator('active', bool)
+    ]
+
+    rules_validator = RuleValidator(allowed_update_types)
+
+    def __init__(self):
+        CRUD.__init__(self, Data, db_handler, CustomerCRUDDataAccess.rules_validator)
+
+customer = CustomerCRUDDataAccess()
+
+customer.create({'name': 'Jon Doe', 'email': 'abc@def.com', 'active': True})
+
+data = result_to_dict(customer.get(1))
+print(data) # {'id': 1, 'name': 'Jon Doe', 'email': 'abc@def.com', 'active': True}
+
+customer.update(1, {'email': 'jon@doe.com'})
+
+data = result_to_dict(customer.get(1))
+print(data) # {'id': 1, 'name': 'Jon Doe', 'email': 'jon@doe.com', 'active': True}
+
+customer.delete(1)
+
+data = customer.get(1) # will raise StatusCodeException Not found
+```
+
+
+### CRUDSoftDeleteDataAccess
+Similar to `CRUDDataAccess` but is used to access and handle soft delete.
+For this to work the database object class must extend `SoftDeleteMixin` to create the required columns for soft delete.
+
+```python
+class CRUDSoftDeleteDataAccess(DataAccess, CRUD):

```

Comment:
it will beter do define all CRUC Data access in a single block on top. 
and specify the parameteres section only once